src/ott/geometry/geodesic.py

```diff
@@ -27,6 +27,38 @@
 __all__ = ["Geodesic"]
 
 
+def compute_dense_laplacian(
+    G: jnp.ndarray, normalize: bool = False
+) -> jnp.ndarray:
+  degree = jnp.sum(G, axis=1)
+  laplacian = jnp.diag(degree) - G
+  if normalize:
+    inv_sqrt_deg = jnp.diag(
+        jnp.where(degree > 0.0, 1.0 / jnp.sqrt(degree), 0.0)
+    )
+    laplacian = inv_sqrt_deg @ laplacian @ inv_sqrt_deg
```

Comment:
Think here we could also use `laplacian = data_inv[:, None] * laplacian * data_inv[None, :]`, should be a bit faster.